### DISCUSSION TITLE
protobufjs and long compatibility

### DISCUSSION THREAD

**Discussion by iamricard** [stephenh/ts-proto#674] [Q&A]
👋🏼 i was wondering if there is or should be documentation around which protobufjs and long versions the produce output is compatible with. i've been assuming that "latest long" and "protobufjs matching ts-proto's version" is the right call, but i'm not sure if that's right?

thanks!

**Comment by stephenh** [comment]
Hey @iamricard ; yeah, that is a good idea. For quite awhile neither protobufjs nor longjs had any major (or really any?) releases, so we didn't really have to worry about it.

But since then, yeah, I think both protobufjs and long have had major version bumps (and unfortunately AFAIU protobufjs 7.x still doesn't support longjs 5.x, so we can't just bump to the latest of each).

Disclaimer that ^ is off the top of my head, but as a lazy ask, if you'd like to submit a PR to the docs that clarifies (and confirms, b/c I'm kinda guessing) the version information, that'd be great! Thanks!

**Reply by iamricard** [reply]
I'll look into submitting a PR for it :)

**Comment by stephenh** [comment]
(Also, we should for sure get around to supporting the latest protobufjs; I have no idea what's involved, but would hope it's not that bad... Another place where PRs are appreciated! :-D)

**Reply by iamricard** [reply]
I'll look into it! I also remember from https://github.com/stephenh/ts-proto/issues/647, that you wanted to look into migrating off of protobufjs/minimal?

A source of inspiration could be https://github.com/protocolbuffers/protobuf-javascript/tree/main/generator

edit:

Having looked at `protobuf-javascript`, it's even more confusing than the `protobufjs` implementation.

linking to the protobuf encoding spec docs for future reference: https://developers.google.com/protocol-buffers/docs/encoding

**Reply by iamricard** [reply]
From the looks of it, the 7.x bump on protobufjs only seemed to affect the compiler. No changes to `protobufjs/minimal` as far as I can tell*:

https://github.com/protobufjs/protobuf.js/blob/master/CHANGELOG.md#700-2022-07-08

And it seems to be compatible with `long@^5`:

https://github.com/protobufjs/protobuf.js/blob/d0268490cb08d2f88e0aafd1d0eb673bd2d3c714/package.json#L54

*the breaking changes are to `converter.js` which is a runtime dep, but not for `minimal.js` (I even bundled `index-minimal.js` with `esbuild` to make sure there were no references to it anywhere).
